### ISSUE TITLE
Pelvis Quaternion Displaying Incorrectly in Python

### ISSUE THREAD

**Issue by owen2t** [microsoft/Azure-Kinect-Sensor-SDK#1760]
I'm trying to visualize the pelvis orientation outputted by the body tracking sdk in python. I'm eventually going to be using the rest of the lower body orientations but I figured it would be best to start with joint 0. 

To calculate the x, y, and z axis of the pelvis, I transform the global kinect orientation vectors [1,0,0], [0,1,0], and [0,0,1] by the quaternion given for the pelvis. I do this using the conjugation equation p'=q * p * q^-1 where q is the quaternion and p is the vector. When I plot these rotated axes over my skeleton (first image) they don't line up with what is shown in the k4a viewer (second image). 

Is there some additional transformation that's required? Or am I misunderstanding how to apply the quaternions? I would really appreciate some guidance. 

(Edit) To sort of rephrase this question, I have found that the navel-pelvis segment and left hip-pelvis segments are always orthogonal. These can be used to create a set of 3 axes that line up with the simple_3d_viewer output (third image). Why does applying the outputted pelvis quaternion to the unit global coordinate axes result in something different/incorrect?

![basic_quat](https://user-images.githubusercontent.com/97454915/160709455-efe4f7dd-4917-4304-b661-19fcb3f989a2.PNG)
![viewer_side](https://user-images.githubusercontent.com/97454915/160709456-cda318d3-0b71-45e7-b1c8-244c28980dd3.PNG)
![pelvis_good](https://user-images.githubusercontent.com/97454915/161333860-25d37f97-a1cd-4c95-83a2-25c9096ba0b6.PNG)

**Comment by owen2t** [comment]
It's now been 2 full weeks since I posted this question. I understand that avateering is one of the focuses of the azure kinect, but there are people and businesses who want to use the kinect for other purposes.

As mentioned in #1414 , there is insufficient documentation on how to actually use the quaternions to get coordinate systems for each body part. I was able to run the kinect in unity and all of the axes defined by the quaternions displayed correctly, so I know it's not something wrong with the body tracking- and I looked through the code for the body tracking sample and there are several transformations being done on the quaternions before they are applied. 

I am asking for some documentation, provided either in the form of an answer to this issue or an addition to the existing set of kinect samples, on how to use the quaterions outside of avateering software. Thank you.

**Comment by Wilberttt** [comment]
Could you please show the Python code to visualize the quaternion?

**Comment by owen2t** [comment]
Sure. First, I apply the joint's quaternion to [1,0,0], [0,1,0], and [0,0,1] with quaternion multiplication. I normalize the resulting vectors and store them in a 3x3 np array in the format [[x vector],[y_vector],[z_vector]]. That array is called xyz. The function that I have to plot takes the origin (for example, the x, y, and z coordinates of the joint) and xyz np array and plots it like this:
`axeslength = 100
x = axeslength * xyz[0,:] + origin
xvec = np.array([x,origin])
y = axeslength * xyz[1,:] + origin
yvec = np.array([y,origin])
z = axeslength * xyz[2,:] + origin
zvec = np.array([z,origin])
ax.scatter(origin[0],origin[1],origin[2],color='black')

ax.plot(xvec[:,0],xvec[:,1],xvec[:,2],color='red',label='x-axis')
ax.plot(yvec[:,0],yvec[:,1],yvec[:,2],color='green',label='y-axis')
ax.plot(zvec[:,0],zvec[:,1],zvec[:,2],color='blue',label='z-axis')`

**Comment by owen2t** [comment]
The application of the quaternion is where something is going wrong and that's what I'm looking for clarity on- the plotting part works fine.

**Comment by Wilberttt** [comment]
You can try axisymmetric inversion of the coordinates provided by the SDK. Compare the flipped image with the one you drew in Python, and you'll get the same result.

**Comment by owen2t** [comment]
I'm not quite sure what you mean by that- Do you mean inverting the x, y, and z values of the joint location coordinates?

**Comment by Wilberttt** [comment]
The image you see in body tracking viewer is mirrored I guess. And the coordinate drwan in Python is correct.

**Comment by owen2t** [comment]
Right. My concern is that the axes generated from the quaternion in the first python image don't match the axes in the body tracking viewer

**Comment by qm13** [comment]
The body tracker viewer mirrors the image.

**Comment by owen2t** [comment]
Okay, thank you for your response. I guess the big question that I have is whether I am taking the right approach to apply the quaternions outputted by the body tracking. I'm happy to attach some code if the explanation of my current method in the original question isn't clear!
